app/super_admin/services.py

```diff
@@ -0,0 +1,162 @@
+    await selectors.get_admin_by_id(admin_id=admin_id, db=db)
+    if db.query(models.AdminConfiguration).filter_by(admin_id=admin_id).first():
+        raise HTTPException(
+            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+            detail=f"admin configuration for admin {admin_id} already exists",
+        )
+    obj = models.AdminConfiguration(admin_id=admin_id)
+    db.add(obj)
+    db.commit()
+    db.refresh(obj)
+    return obj
+
+
+async def create_admin_notification(admin_id: int, content: str, db: Session):
+    """This function creates an admin notification
+
+    Args:
+        admin_id (int): The admin's ID
+        content (str): The notification content
+        db (Session): The database session
+
+    Returns:
+        models.AdminNotification: The created notifcation notification obj
+    """
+    await selectors.get_admin_by_id(admin_id=admin_id, db=db)
+    obj = models.AdminNotification(admin_id=admin_id, content=content)
+    db.add(obj)
+    db.commit()
+    db.refresh(obj)
+    return obj
+
+
+async def login_admin(
+    data: base_schemas.AdminLoginCredential,
+    db: Session,
+    raise_exception: bool = True,
+):
+    """This function logs in an admin and updates their last_login
+
+    Args:
+        data (base_schemas.AdminLoginCredential): The admin's login credentials
+        db (Session): The database session
+        raise_exception(bool, default=True): Indicates whether to raise an exception or not
+
+    Raises:
+        HTTPException[401]: Invalid login credentials
+
+    Returns:
+        (models.Admin, None): The admin's model obj
+    """
+    admin = await selectors.get_admin_by_email(
+        email=data.email, db=db, raise_exception=False
+    )
+    if not admin:
+        if raise_exception:
+            raise HTTPException(
+                status_code=status.HTTP_401_UNAUTHORIZED,
+                detail="Invalid login credentials",
+            )
+        return None
+    if verify_password(plain_password=data.password, hashed_password=admin.password):
+        admin.last_login = datetime.now()
+        db.commit()
+        db.refresh(admin)
+        return admin
+    if raise_exception:
+        raise HTTPException(
+            status_code=status.HTTP_401_UNAUTHORIZED, detail="Invalid login credentials"
+        )
+    return None
+
+
+async def create_admin_refresh_token(admin_id: int, token: str, db: Session):
+    """This function creates an admin refresh token
+
+    Args:
+        admin_id (int): The admin's ID
+        token(str): The access token
+        db (Session): The database session
+
+    Returns:
+        models.AdminRefreshToken: The created admin refresh token obj
+    """
+    await selectors.get_admin_by_id(admin_id=admin_id, db=db)
+    obj = models.AdminRefreshToken(admin_id=admin_id, token=token)
+    db.add(obj)
+    db.commit()
+    db.refresh(obj)
+    return obj
+
+
+async def edit_admin(admin_id: int, data: edit_schemas.AdminEdit, db: Session):
```

Comment:
Why are you adding admin permissions to schema and you aren't checking who is changing what, add a check to make sure only super admins can edit permissions